clang/lib/Frontend/TextDiagnosticPrinter.cpp

```diff
@@ -133,12 +133,16 @@ void TextDiagnosticPrinter::HandleDiagnostic(DiagnosticsEngine::Level Level,
   // diagnostics in a context that lacks language options, a source manager, or
   // other infrastructure necessary when emitting more rich diagnostics.
   if (!Info.getLocation().isValid()) {
```

Comment:
So the problem you're describing only exists for diagnostics without valid locations? :thinking:

Reply:
No, it exists for all of them. The handling is a bit different for diagnostics without valid locations, so I had to modify this place and `TextDiagnostic` class to handle the case with valid locations.